provider/script.go

```diff
@@ -16,15 +16,19 @@ var ScriptCRONParser = cron.NewParser(cron.Second | cron.Minute | cron.Hour | cr
 func scriptResource() *schema.Resource {
 	return &schema.Resource{
 		Description: "Use this resource to run a script from an agent.",
-		CreateContext: func(ctx context.Context, rd *schema.ResourceData, i interface{}) diag.Diagnostics {
+		CreateContext: func(_ context.Context, rd *schema.ResourceData, _ interface{}) diag.Diagnostics {
 			rd.SetId(uuid.NewString())
 			runOnStart, _ := rd.Get("run_on_start").(bool)
+			startBlocksLogin, _ := rd.Get("start_blocks_login").(bool)
 			runOnStop, _ := rd.Get("run_on_stop").(bool)
 			cron, _ := rd.Get("cron").(string)
 
 			if !runOnStart && !runOnStop && cron == "" {
 				return diag.Errorf("at least one of run_on_start, run_on_stop, or cron must be set")
 			}
+			if !runOnStart && startBlocksLogin {
+				return diag.Errorf("start_blocks_login can only be set if run_on_start is true")
+			}
```

Comment:
Does this extra check risk breaking any existing usage?

Reply:
Possibly, but it also alerts users to a misconfiguration that will cause unexpected results.

Reply:
Might be worth calling this out as a potentially breaking change then. LGTM otherwise 👍

Reply:
I broke it out to a separate PR, this was a good callout tbh: https://github.com/coder/terraform-provider-coder/pull/192